### PR TITLE
Added Cirrus CI and Jest

### PR DESCRIPTION
**Pull request by RDIL**
This is still a work in progress

### CODE REVIEW

**Comment by N8python** [comment]
Ok - thx

**Comment by N8python** [comment]
What about the other test - there were two tests in the /test directory? I looked in your fork and there was only one test in the /compiler directory

File: src/compiler/gen.js [hunk 1]
```diff
@@ -604,14 +604,14 @@ ${type}.${key} = ${f};
...
-function generateParent(type, parts, static = {}) {
+function generateParent(type, parts, static_item = {}) {
   let res = (
     `let ${type} = {
   order: ${zStringify(parts)},
   ${parts.join(",\n\t")}
 };`
   );
-  res += generateStatics(type, static);
+  res += generateStatics(type, static_item);
```
Comment (RDIL) [inline src/compiler/gen.js]: had to make this change as the name conflicted with the modifier

**Comment by RDIL** [comment]
Not sure what happened to the other test - going to try to readd it

File: src/compiler/gen.js (hunk 1 above)
Comment (N8python) [reply src/compiler/gen.js]: yeah - that's fine!

**Comment by RDIL** [comment]
@N8python one more step is required - you will need to get the free plan for the z organization on https://github.com/marketplace/cirrus-ci , make sure to tick all repos on the install page

**Comment by N8python** [comment]
Ok - Thx!

**Pull request merged**